app/models/daos/PasswordInfoDAO.scala

```diff
@@ -0,0 +1,101 @@
+package models.daos
+
+import com.mohiva.play.silhouette.api.LoginInfo
+import com.mohiva.play.silhouette.api.util.PasswordInfo
+import com.mohiva.play.silhouette.impl.daos.DelegableAuthInfoDAO
+import play.api.libs.concurrent.Execution.Implicits._
+import javax.inject.Inject
+import play.api.libs.concurrent.Execution.Implicits._
+import play.api.db.slick.DatabaseConfigProvider
+import scala.concurrent.Future
+
+/**
+ * The DAO to store the password information.
+ */
+class PasswordInfoDAO @Inject() (protected val dbConfigProvider: DatabaseConfigProvider)
+    extends DelegableAuthInfoDAO[PasswordInfo] with DAOSlick {
+
+  import driver.api._
+
+  protected def passwordInfoQuery(loginInfo: LoginInfo) = for {
+    dbLoginInfo <- loginInfoQuery(loginInfo)
+    dbPasswordInfo <- slickPasswordInfos if dbPasswordInfo.loginInfoId === dbLoginInfo.id
+  } yield dbPasswordInfo
+  
+  // Use subquery workaround instead of join to get authinfo because slick only supports selecting
+  // from a single table for update/delete queries (https://github.com/slick/slick/issues/684).
+  protected def passwordInfoSubQuery(loginInfo: LoginInfo) =
+    slickPasswordInfos.filter(_.loginInfoId in loginInfoQuery(loginInfo).map(_.id))
+
+  protected def addAction(loginInfo: LoginInfo, authInfo: PasswordInfo) =
+    loginInfoQuery(loginInfo).result.head.flatMap { dbLoginInfo =>
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Usage of head on collections.](https://www.codacy.com/app/HiP-App/HiPCMS/file/2283447077/issues/source?bid=3144114&fileBranchId=3144125#l31)